app/code/core/Mage/Admin/Model/Acl/Assert/Ip.php

```diff
@@ -37,18 +37,24 @@ class Mage_Admin_Model_Acl_Assert_Ip implements Zend_Acl_Assert_Interface
     /**
      * Check whether ip is allowed
      *
-     * @param Zend_Acl $acl
-     * @param Zend_Acl_Role_Interface $role
-     * @param Zend_Acl_Resource_Interface $resource
+     * @param Mage_Admin_Model_Acl $acl
+     * @param Mage_Admin_Model_Acl_Role $role
+     * @param Mage_Admin_Model_Acl_Resource $resource
      * @param string $privilege
      * @return boolean
      */
-    public function assert(Mage_Admin_Model_Acl $acl, Mage_Admin_Model_Acl_Role $role = null,
-                           Mage_Admin_Model_Acl_Resource $resource = null, $privilege = null)
-    {
+    public function assert(
+        Mage_Admin_Model_Acl $acl,
+        Mage_Admin_Model_Acl_Role $role = null,
+        Mage_Admin_Model_Acl_Resource $resource = null,
+        $privilege = null
+    ) {
         return $this->_isCleanIP(Mage::helper('core/http')->getRemoteAddr());
     }
 
+    /**
+     * @param bool $ip
```

Comment:
this should be string as Mage::helper('core/http')->getRemoteAddr() returns string

Reply:
maybe return should be bool

Reply:
Mhhh ... this methode is empty ....  How abaout ...

- `@param`: `Mage::helper('core/http')->getRemoteAddr()` returns `string` or `false` ... 
- `@return void`

```
    protected function _isCleanIP($ip)
    {
        // ...
    }
```

Reply:
ok, seems the classes are not used in the core anyway